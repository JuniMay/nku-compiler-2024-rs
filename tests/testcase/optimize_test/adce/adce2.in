65535
1000000