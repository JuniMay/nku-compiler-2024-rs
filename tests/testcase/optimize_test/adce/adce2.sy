int main()
{
    int p = getint();
    int n = getint();
    int S = 0;
    int i = 0;
    int i1=0,i2=1,i3=2,i4=3,i5=0,i6=0,i7=0,i8=10,i9=0,i10=0,i11=0,i12=0,i13=0,i14=0,i15=0,i16=0,i17=0,i18=0,i19=0,i20=0;
    while(i < n){
        S = S + i + i/4 + i/10;
        S = S % 998244353;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1+1;
        i2=i2+1;
        i3=i3+1;
        i4=i4+1;
        i5=i5+1;
        i6=i6+1;
        i7=i7+1;
        i8=i8+1;
        i9=i9+1;
        i10=i10+1;
        i11=i11+1;
        i12=i12+1;
        i13=i13+1;
        i14=i14+1;
        i15=i15+1;
        i16=i16+1;
        i17=i17+1;
        i18=i18+1;
        i19=i19+1;
        i20=i20+1;
        i1=i1*2;
        i2=i2*2;
        i3=i3*2;
        i4=i4*2;
        i5=i5*2;
        i6=i6*2;
        i7=i7*2;
        i8=i8*2;
        i9=i9*2;
        i10=i10*2;
        i11=i11*2;
        i12=i12*2;
        i13=i13*2;
        i14=i14*2;
        i15=i15*2;
        i16=i16*2;
        i17=i17*2;
        i18=i18*2;
        i19=i19*2;
        i20=i20*2;
        if(i%2 == 0){
            i1=i1/2;
            i2=i2/2;
            i3=i3/2;
            i4=i4/2;
            i5=i5/2;
            i6=i6/2;
            i7=i7/2;
            i8=i8/2;
            i9=i9/2;
        }
        i1=i1%998244353;
        i2=i2%998244353;
        i3=i3%998244353;
        i4=i4%998244353;
        i5=i5%998244353;
        i6=i6%998244353;
        i7=i7%998244353;
        i8=i8%998244353;
        i9=i9%998244353;
        i10=i10%998244353;
        i11=i11%998244353;
        i12=i12%998244353;
        i13=i13%998244353;
        i14=i14%998244353;
        i15=i15%998244353;
        i16=i16%998244353;
        i17=i17%998244353;
        i18=i18%998244353;
        i19=i19%998244353;
        i20=i20%998244353;
        i1=i1%p;
        i2=i2%p;
        i3=i3%p;
        i4=i4%p;
        i5=i5%p;
        i6=i6%p;
        i7=i7%p;
        i8=i8%p;
        i9=i9%p;
        i10=i10%p;
        i11=i11%p;
        i12=i12%p;
        i13=i13%p;
        i14=i14%p;
        i15=i15%p;
        i = i + 1;
        i16=i16%p;
        i17=i17%p;
        i18=i18%p;
        i19=i19%p;
        i20=i20%p;
    }
    putint(S);
    putch(10);
    return 0;
}