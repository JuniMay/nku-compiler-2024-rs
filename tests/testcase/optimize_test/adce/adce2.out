185317372
0
